scripts/ci/generate_install_lists.py

```diff
@@ -58,33 +58,35 @@ def main(argv=sys.argv[1:]):
     os.chdir(workspace_root)
 
     with Scope('SUBSECTION', 'mark packages with IGNORE files'):
-        packages = locate_packages(workspace_root)
+        all_packages = locate_packages(workspace_root)
         if args.package_selection_args:
             print(
                 'Using package selection arguments:',
                 args.package_selection_args)
             selected_packages = locate_packages(
                 workspace_root, extra_args=args.package_selection_args)
 
-            to_ignore = packages.keys() - selected_packages.keys()
+            to_ignore = all_packages.keys() - selected_packages.keys()
             print('Ignoring %d packages' % len(to_ignore))
             for package in sorted(to_ignore):
                 print('-', package)
-                package_root = packages.pop(package)
+                package_root = all_packages[package]
                 Path(package_root, 'COLCON_IGNORE').touch()
 
         print('There are %d packages which meet selection criteria' %
-              len(packages))
+              len(selected_packages))
```

Comment:
Yes - `packages` and `selected_packages` should have been the same after the loop prior to this change.

Rather than re-enumerating the packages later, I just modified the code to keep the list of packages around for later use.